macosx/Controller.mm

```diff
@@ -122,7 +122,7 @@ typedef NS_ENUM(unsigned int, sortOrderTag) { //
 
 static void altSpeedToggledCallback([[maybe_unused]] tr_session* handle, bool active, bool byUser, void* controller)
 {
-    NSDictionary* dict = [[NSDictionary alloc] initWithObjects:@[ @(active), @(byUser) ] forKeys:@[ @"Active", @"ByUser" ]];
+    NSDictionary* dict = @{@"Active" : @(active), @"ByUser" : @(byUser)};
```

Comment:
This wasn't autoreleased before. Have you checked that it won't crash here?

Reply:
I don't think I understand what you mean here.
[Clang Doc](https://clang.llvm.org/docs/ObjectiveCLiterals.html#container-literals) states this 2 forms are equivalent.
There were no retention annotations before.

Reply:
By the way, I see no crash while toggling altMode, `p dict.retainCount` evaluates to 3 in LLDB

Reply:
Also, shile profiling with `Instruments.app` no leaks detected after toggling altMode like ~100 times.
<img width="797" alt="image" src="https://user-images.githubusercontent.com/8330119/175841167-8bbabdd4-51ca-44ef-b119-727a02024e3d.png">

Reply:
They're not equivalent - the `init` form returns the object with a retain count of 1, while the literal form returns the autoreleased object (equivalent to the `arrayWithObjects` method of the class). This caused crashes before when I migrated Transmission to ARC, so I just wanted to make sure that you'd tested this. Seems that it works without issues, so I mark it as approved!

Reply:
@DevilDimon OK to mark this conversation as resolved?

Reply:
@DevilDimon in ARC, they are equivalent. The difference only arises in non-ARC.

Reply:
@Coeur not really equivalent - autoreleased objects are only released upon the drain of their respective autorelease pool. If you create tons of objects in a tight loop it's better to create them using the `init` initialiser rather than the `arrayWith` one to conserve RAM.